parquet/src/arrow/async_reader/mod.rs

```diff
@@ -111,15 +111,11 @@ pub trait AsyncFileReader: Send {
     /// Provides asynchronous access to the [`ParquetMetaData`] of a parquet file,
     /// allowing fine-grained control over how metadata is sourced, in particular allowing
     /// for caching, pre-fetching, catalog metadata, decrypting, etc...
-    ///
-    /// By default calls `get_metadata()`
+    //  No default here because this is too easy to get buggy behavior
     fn get_metadata_with_options<'a>(
         &'a mut self,
         options: &'a ArrowReaderOptions,
-    ) -> BoxFuture<'a, Result<Arc<ParquetMetaData>>> {
-        let _ = options;
-        self.get_metadata()
-    }
+    ) -> BoxFuture<'a, Result<Arc<ParquetMetaData>>>;
```

Comment:
@rok This default was actually very painful for me. There was a datafusion class that was missing an implementation of get_metadata_with_options which was causing decryption to fail. I could NOT figure out what was wrong since everything looked fine and the decryption setting were there. Given how easy this is to miss, I think it is better to force an implementation / get a compiler warning.

Reply:
The problem is this is a public trait that consumers of arrow-rs might implement, so it would be a breaking change to not provide a default implementation. We might be able to do that but it might be tough to convince others that it's necessary.

Reply:
I think you make a good point but this has already been merged and released, so removing the default should probably be a separate PR and discussion to the encryption changes.

Reply:
OK. Change backed out.